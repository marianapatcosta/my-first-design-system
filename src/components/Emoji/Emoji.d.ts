import React, { CSSProperties } from 'react';
export interface EmojiProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * element's label'
     */
    label?: string;
    /**
     * emoji to display
     */
    emoji: any;
}
export declare const Emoji: React.FC<EmojiProps>;
