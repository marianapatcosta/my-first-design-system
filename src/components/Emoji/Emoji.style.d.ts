export declare const StyledEmoji: import("styled-components").StyledComponent<"div", any, {}, never>;
