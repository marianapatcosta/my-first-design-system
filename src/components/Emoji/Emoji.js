var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx } from "react/jsx-runtime";
import { StyledEmoji } from './Emoji.style';
export var Emoji = function (_a) {
    var label = _a.label, emoji = _a.emoji;
    return (_jsx(StyledEmoji, __assign({ role: 'img', "aria-label": label ? label : '', "aria-hidden": label ? 'false' : 'true' }, { children: emoji }), void 0));
};
Emoji.defaultProps = {
    id: '',
    className: '',
    style: {},
    label: '',
    emoji: null,
};
