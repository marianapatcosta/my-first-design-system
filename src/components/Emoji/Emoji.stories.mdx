import { Meta } from "@storybook/addon-docs/blocks";
import { Emoji } from "./Emoji";

<Meta title="Design System|Emoji" component=Emoji />

# Emoji
Emoji component that avoid ESLint errors existing when using emoji by only 'copy and paste'

## Usage
Use emoji to display an emoji

- Default Emoji:
  <Story name='Default'>
    <Emoji emoji='ðŸ˜ƒ' label= 'smiley'/>
  </Story>


## Other important notes
not applicable.