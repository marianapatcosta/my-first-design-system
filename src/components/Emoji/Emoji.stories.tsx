import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Emoji, EmojiProps } from './Emoji';

export default {
  title: 'Emoji',
  component: Emoji,
  // args behavior in each story
  args: {},
} as Meta;

const Template: Story<EmojiProps> = (args) => <Emoji {...args} />;

export const Default = Template.bind({});
Default.args = {
  emoji: 'ðŸ˜ƒ',
  label: 'smiley',
};
