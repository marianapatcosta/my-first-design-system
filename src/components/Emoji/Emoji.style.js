var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
import { baseStyles } from '../../styles';
export var StyledEmoji = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n  font-size: 90%;\n"], ["\n  ", "\n  font-size: 90%;\n"])), baseStyles);
var templateObject_1;
