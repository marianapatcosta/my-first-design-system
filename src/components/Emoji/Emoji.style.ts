import styled from 'styled-components';
import { baseStyles } from '../../styles';

export const StyledEmoji = styled.div`
  ${baseStyles}
  font-size: 90%;
`;
