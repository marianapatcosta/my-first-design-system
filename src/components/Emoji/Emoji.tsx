import React, { CSSProperties } from 'react';
import { StyledEmoji } from './Emoji.style';

export interface EmojiProps {
  /**
   * element id
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * additional style
   */
  style?: CSSProperties;
  /**
   * element's label'
   */
  label?: string;
  /**
   * emoji to display
   */
  emoji: any;
}

export const Emoji: React.FC<EmojiProps> = ({ label, emoji }) => {
  return (
    <StyledEmoji
      role='img'
      aria-label={label ? label : ''}
      aria-hidden={label ? 'false' : 'true'}
    >
      {emoji}
    </StyledEmoji>
  );
};

Emoji.defaultProps = {
  id: '',
  className: '',
  style: {},
  label: '',
  emoji: null,
};
