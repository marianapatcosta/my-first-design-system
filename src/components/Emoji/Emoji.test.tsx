import React from 'react';
import { render } from '@testing-library/react';
import { Emoji } from './Emoji';

describe('Emoji', () => {
  test('renders without errors and matches snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Emoji emoji='ðŸ˜ƒ' />);

    expect(firstChild).toMatchSnapshot();
  });

  test('display label prop as aria-label', () => {
    const { getByLabelText } = render(<Emoji emoji='ðŸ˜ƒ' label='smiley' />);

    const emoji = getByLabelText(/smiley/i) as HTMLInputElement;
    expect(emoji).toBeTruthy();
  });
});
