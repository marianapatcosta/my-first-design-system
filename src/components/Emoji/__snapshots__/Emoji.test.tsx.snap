// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji renders without errors and matches snapshot 1`] = `
<div
  aria-hidden="true"
  aria-label=""
  class="sc-bdfBwQ bTxJHq"
  role="img"
>
  ðŸ˜ƒ
</div>
`;
